src/pages/welcome.tsx

```diff
@@ -471,6 +483,353 @@ const DataCheckSection = ({
+                    // info from https://covid19.trackvaccines.org/country/thailand/
+                    options={[
+                      { value: "comirnaty", label: "Comirnaty (Pfizer)" },
+                      { value: "coronavac", label: "CoronaVac (Sinovac)" },
+                      { value: "vaxzevria", label: "Vaxzevria (AstraZeneca)" },
+                      { value: "spikevax", label: "Spikevax (Moderna)" },
+                      { value: "jcovden", label: "Jcovden (J&J)" },
+                      { value: "covilo", label: "Covilo (Sinopharm)" },
+                      {
+                        value: "covovax",
+                        label: "COVOVAX (Novavax formulation)",
+                      },
+                    ]}
+                    onChange={(e: string) => setForm({ ...form, provider: e })}
+                    defaultValue={"pfizer"}
+                  />
+                </div>
+                <div className="layout-grid-cols-2 !gap-y-0">
+                  <KeyboardInput
+                    name="vaccine-lot"
+                    type="text"
+                    label={t("vaccine.lot.label", { ns: "covid" })}
+                    onChange={(e) => setForm({ ...form, lotno: e })}
+                    attr={{ disabled }}
+                  />
+                  <KeyboardInput
+                    name="vaccine-administering-center"
+                    type="text"
+                    label={t("vaccine.administeringCenter.label", {
+                      ns: "covid",
+                    })}
+                    onChange={(e) =>
+                      setForm({ ...form, administeringCenter: e })
+                    }
+                    attr={{ disabled }}
+                  />
+                </div>
+              </section>
+              <CardActions>
+                <Button
+                  label={t("welcome.vaccineData.action.add")}
+                  type="filled"
+                  onClick={async () => {
+                    const { data: personid } = await getPersonIDFromUser(
+                      supabase,
+                      user!
+                    );
+                    toggleLoading();
+                    await addVaccineRecord(
+                      supabase,
+                      {
+                        id: 0,
+                        doseNo: 0,
+                        lotNo: form.lotno,
+                        vaccineDate: new Date(form.date),
+                        administeredBy: form.administeringCenter,
+                        vaccineName: form.provider,
+                      },
+                      personid!
+                    );
+                    const newVaccineRecords = await getVaccineRecordbyPersonId(
+                      supabase,
+                      personid!
+                    );
+                    toggleLoading();
+                    setVaccineData(newVaccineRecords);
+                  }}
```

Comment:
Don’t you agree that altering the database here makes things much more complicate (for us) and slow (for users)? Why not just keep track of everything in states and just do all the changes afterwards?

Reply:
If you ask me as a user, I would expect when I click add the data should reflect on the database the same way as it does on the screen. For the complicated part, I want to make sure any data being display is accurate to the backend so I just call the backend to get the data. If speed really become an issue then we could optout manually add the new record to the state then resort everything to get the actual dose number.

Reply:
But it literally says “Save and continue,” though, implying that your changes are not saved yet. The fact that the changes are saved confused me (I noticed from the delay in deleting) so I thought to check the code.
I still don’t understand how this is a better approach for anyone. Working with the backend sucks and I’m pretty sure you’re supposed to contact the backend as few times as possible. Anyway, I know you want to get this out fast so we’ll just leave this for the future.